packages/odyssey-react-mui/src/RadioGroup.tsx

```diff
@@ -73,31 +89,79 @@ const RadioGroup = ({
   HintLinkComponent,
   id: idOverride,
   isDisabled,
+  isReadOnly = false,
   label,
   name: nameOverride,
   onChange: onChangeProp,
   testId,
   translate,
   value,
 }: RadioGroupProps) => {
+  const [internalValue, setInternalValue] = useState(value ?? defaultValue);
+
   const controlledStateRef = useRef(
     getControlState({
       controlledValue: value,
       uncontrolledValue: defaultValue,
     }),
   );
+
   const inputValues = useInputValues({
     defaultValue,
     value,
     controlState: controlledStateRef.current,
   });
 
+  const isControlled =
+    controlledStateRef.current === ComponentControlledState.CONTROLLED;
+
   const onChange = useCallback<NonNullable<MuiRadioGroupProps["onChange"]>>(
-    (event, value) => {
-      onChangeProp?.(event, value);
+    (event, newValue) => {
+      if (!isReadOnly) {
+        if (!isControlled) {
+          setInternalValue(newValue);
+        }
+        onChangeProp?.(event, newValue);
+      }
     },
-    [onChangeProp],
+    [onChangeProp, isReadOnly, isControlled],
   );
+
+  const handleReadOnlyClick = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      if (isReadOnly) {
+        event.preventDefault();
+      }
+    },
+    [isReadOnly],
+  );
+
+  const memoizedChildren = useMemo(
+    () =>
+      React.Children.map(children, (child) => {
+        if (React.isValidElement<ExtendedRadioProps>(child)) {
+          return React.cloneElement(child, {
+            isReadOnly,
+            isDisabled,
+            isChecked:
+              (isControlled ? inputValues.value : internalValue) ===
+              child.props.value,
+            onReadOnlyClick: handleReadOnlyClick,
```

Comment:
@benlister-okta This isn't a valid prop is it? My assumption is this doesn't do anything and probably passes this prop into the DOM.

Reply:
It isn't a valid prop in `RadioGroup`, but the component uses it to pass `isChecked` to the children Radio components in the group to determine which one to check.

Reply:
@benlister-okta I pulled your branch down, still confused as to what this does. I removed it entirely and everything seems to work fine. Also, before removing, I put a log in the `handleReadOnlyClick` function and it never gets called

Reply:
Per our Slack convo: We do not need to pass down the `onReadOnlyClick` prop. It wasn't firing at all and the onClick handler in `Radio` already prevents default behavior in the `isReadOnly` case.

Also, from what I can tell, we do not need to pass `isChecked` either